From 26dc643d0067db6ed8bf4b8a26b4b31115619c9f Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Sun, 16 Feb 2014 23:52:10 +0100
Subject: [PATCH] add Breaks against obsolete packages using install-info in
 their prerm

These packages have been removed after they were released with lenny or
squeeze, but they may survive distupgrades to wheezy and jessie.
Since their prerm uses install-info they cannot be removed any longer
after installing dpkg 1.17.x and there is no upgrade path to a fixed
package version, so add Breaks against them.  (Closes: #735159)

All Breaks have the form PACKAGE (<< VERSION).
If the source package still exists, I used the VERSION that dropped the
problematic PACKAGE. Otherwise I used the next possible maintainer
upload VERSION that could reintroduce PACKAGE into sid.
---
 debian/changelog |  7 ++++++-
 debian/control   | 40 +++++++++++++++++++++++++++++++++++++++-
 2 files changed, 45 insertions(+), 2 deletions(-)

diff --git a/debian/changelog b/debian/changelog
index 3128c68..51d370f 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,5 +1,10 @@
 dpkg (1.17.7) UNRELEASED; urgency=low
 
+  [ Andreas Beckmann ]
+  * dpkg: Add Breaks against packages using install-info that were removed
+    after lenny or squeeze, but may have survived upgrades to wheezy.
+    (Closes: #735159)
+
   [ Updated dpkg translations ]
   * German (Sven Joachim).
   * Swedish (Peter Krefting).
@@ -11,7 +16,7 @@ dpkg (1.17.7) UNRELEASED; urgency=low
   [ Updated manpages translations ]
   * German (Helge Kreutzmann).
   * Swedish (Peter Krefting).
- 
+
  -- Guillem Jover <guillem@debian.org>  Wed, 15 Jan 2014 19:02:55 +0100
 
 dpkg (1.17.6) unstable; urgency=low
diff --git a/debian/control b/debian/control
index 8cd022d..9f87658 100644
--- a/debian/control
+++ b/debian/control
@@ -35,7 +35,45 @@ Essential: yes
 Pre-Depends: ${shlibs:Depends}, tar (>= 1.23)
 Depends: ${misc:Depends}
 Breaks: dpkg-dev (<< 1.15.8), libdpkg-perl (<< 1.15.8),
- apt (<< 0.7.7), aptitude (<< 0.4.7-1)
+ apt (<< 0.7.7), aptitude (<< 0.4.7-1),
+# the following Breaks against removed packages using install-info
+# in their prerm script can be removed in jessie+1
+# these packages have been removed after squeeze:
+ ggz-docs (<< 0.0.14.1-2),
+ glame (<< 2.0.1-6),
+ gtalk (<< 0.99.10-16),
+ libalogg-dev (<< 1.3.7-2),
+ r6rs-doc (<< 1.0-2),
+ ulog-acctd (<< 0.4.3-3),
+# these non-free packages have been removed after squeeze:
+ cpp-4.1-doc (<< 4.1.2.nf2-4),
+ gcc-4.1-doc (<< 4.1.2.nf2-4),
+ gcj-4.1-doc (<< 4.1.2.nf2-4),
+ gfortran-4.1-doc (<< 4.1.2.nf2-4),
+ gnat-4.1-doc (<< 4.1.2.nf2-4),
+ cpp-4.2-doc (<< 4.2.4.nf1-4),
+ gcc-4.2-doc (<< 4.2.4.nf1-4),
+ gcj-4.2-doc (<< 4.2.4.nf1-4),
+ gfortran-4.2-doc (<< 4.2.4.nf1-4),
+ gnat-4.2-doc (<< 4.2.4.nf1-4),
+# these packages have been removed after lenny:
+ ada-reference-manual (<< 20021112web-4),
+ asn1-mode (<< 2.7-7),
+ bogosort (<< 0.4.2-3),
+ cl-yacc (<< 0.3-3),
+ libgtk1.2-doc (<< 1.2.10-19),
+ libnettle-dev (<< 2),
+ liborbit-dev (<< 0.5.17-12),
+ libreadline5-dev (<< 5.2-8),
+ librep-doc (<< 0.90),
+ mmucl (<< 1.5.2-3),
+ nxml-mode (<< 20041004-9),
+ serveez-doc (<< 0.1.5-3),
+ slat (<< 2.0-6),
+ texlive-base-bin-doc (<< 2007.dfsg.2-9),
+ ttcn-el (<< 0.6.9-2),
+ xconq-doc (<< 7.4.1-5),
+ zenirc (<< 2.112.dfsg-1),
 Suggests: apt
 Replaces: manpages-it (<< 2.80-4)
 Description: Debian package management system
-- 
1.8.5.3

